### DISCUSSION TITLE
Ethernet driver remains in Stopped status, no additional connection listed under Network Interface

### DISCUSSION THREAD

**Discussion by hetzjagd** [bb-qq/r8152#310] [Q&A] [answered]
I'm using a DS918+ with DSM 7.1.1-42962 Update 5 and the CableCreation 2.5G Ethernet to USB Adapter (Type-A).  I have plugged it into the front USB3 port.  I have gone through the steps and restarted the NAS but I seem to be stuck.  

Everything seems to happen as per the steps, installing from the .spk file gives a fail error, running the ssh command seems to be accepted happily, then if I understand the instructions I attempted installing from .spk file again before restarting.  Upon restarting the driver still shows as stopped in the list of Packages.

I'm using the r8152-apollolake-2.16.3-4_7.2.spk because of my NAS model and DSM version.

I have also tried the back usb port but I get the same result.  If I plug a data usb drive into the front usb port it mounts as usbshare1 and is accessible, so I think the port is fine.

Here's a screenshot of my DSM packages https://imgur.com/a/fiMgc2V

I have had a look through https://github.com/bb-qq/r8152/wiki/Troubleshooting but I don't see anything that would apply to my situation.  My adapter is Type-A so there is no extension cable or additional connector used.  Other people with DS918+ have said the front port works, or is the best.  Certainly when I plug it in the device lights up so the ports don't seem to be dead.  I have no other usb devices attached to the NAS when I am trying this and normally I don't have any attached anyway.  As far as using the latest drivers and firmware go, I have the latest on my router but I don't know if that matters yet if the ethernet adapter device doesn't even show up as a network interface in DSM.  Everything else I can see on the Troubleshooting page seems related to configuration after a successful install of the driver, which is where I'm stuck.

I note that I am unable to navigate to /opt/sbin/spk_su to check if the package files are being downloaded there.  When I login via winscp it gives me permission denied just at /opt/ yet I'm using an account with admin permission and otherwise can run ssh commands and do anything else I can think of that would require admin permissions.  The default DSM admin account is currently disabled and has been for a long time.

I also note that the Cable Creation ethernet adapter I received from Amazon Australia has Product Number: CD0673 on the packet.  However my adapter looks less wide than the one shown in the photos in the Amazon US linked product page https://amzn.to/39yfZyj  My one is not much wider than the ethernet port itself but the Amazon US one shows something wider than that.  My one does not appear to be openable (no seams to pry apart) at all so I cannot confirm the chipset inside without risking damaging the outer casing.  The page for Amazon Australia also shows the wider adapter in the product photos https://www.amazon.com.au/CableCreation-Gigabit-Ethernet-Compatible-10-6-10-15/dp/B07VNFLTLD/ref=sr_1_2?__mk_ja_JP=%E3%82%AB%E3%82%BF%E3%82%AB%E3%83%8A&keywords=cable%2Bcreation%2B2.5Gbps%2Bethernet&qid=1686007242&sr=8-2&th=1 however the description would suggest it's the same model and same chipset RTL8156.

And for good measure I have attached an ethernet cable from the adapter to my router to see if maybe that was all that was needed but of course, as there is no network interface for it showing in my DSM, that didn't do anything.

When the adapter is plugged into any of the usb ports both a white and orange light come on on the device, separate led's.  Plugging in an ethernet cable the white light goes out and the orange remains lit.

If I plug the adapter into a Windows 10 PC and install the necessary driver it works without an issue in fact I've updated this post now while connected to it.

I'm using Firefox, not Safari.

dmesg output https://pastebin.com/vDxXkvj2
ifconfig -a output https://pastebin.com/SxVGM8S1
lsusb output https://pastebin.com/912SnvfK

The contents of my /var/packages/r8152/var/log/start-stop-status.log can be found at https://pastebin.com/cuSkeGVD

I note the errors (cut from the log, they don't appear right next to each other like below)

insmod: ERROR: could not insert module /volume1/@appstore/r8152/r8152/r8152.ko: Operation not permitted
rmmod: ERROR: Module r8152 is not currently loaded

If I try run these commands 

sudo /var/packages/r8152/scripts/start-stop-status stop
sudo /var/packages/r8152/scripts/start-stop-status start

It gives an error on the second command; I've shortened it to what just seems important below but the full result is visible here https://pastebin.com/v3mq7P4x

+ '[' -e /var/packages/r8152/target/r8152/mii.ko ']'
+ '[' -e /var/packages/r8152/target/r8152/usbnet.ko ']'
+ /sbin/insmod /var/packages/r8152/target/r8152/r8152.ko
insmod: ERROR: could not insert module /var/packages/r8152/target/r8152/r8152.ko: Invalid module format

https://pastebin.com/v3mq7P4x

For what it's worth, I'm trying for this adapter to work because I'm concerned the NIC in the NAS is dying (one port dead, the other port 100Mbps connection max) and if it gets any worse maybe I won't be able to access the NAS at all.  The extra speed would be nice but I'm just trying to save this things life.

I'm thinking the next step is to downgrade the DSM version if that's possible but I would like confirmation that that is the next thing I should try.  I'm also not sure if I'm capable of the downgrade as a brief look suggests I will lose my data "Keep in mind that at the moment, data loss is unavoidable as part of this process!" https://www.blackvoid.club/dsm-7-to-dsm-6-downgrade/ which I assume is referring to the primary data on the disks and not just DSM configuration and packages.  If it's the latter then I don't care about starting again, but I don't know if I can afford to wipe everything and download ~14tb from my google drive backup.  If it's necessary however..

**Comment by hetzjagd** [comment, marked as answer]
I've made a grave mistake.  I needed the 7.1 version of the driver, not 7.2.  This whole thread can be deleted, it won't help anyone.

**Reply by conman33158** [reply]
Just on a side note, the latest "Version: 7.2-64570 Update 1" will remove the RTL driver and will need to be re-installed, well at least it did on my DS718+ 👍
